internal/pkg/agent/cmd/validate.go

```diff
@@ -0,0 +1,40 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/internal/pkg/cli"
+	"github.com/elastic/elastic-agent/internal/pkg/otel"
+)
+
+func newValidateCommandWithArgs(_ []string, _ *cli.IOStreams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "validate",
+		Short:         "Validates the OpenTelemetry collector configuration without running the collector",
+		SilenceUsage:  true, // do not display usage on error
+		SilenceErrors: true,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			ctx := cmd.Context()
+			cfgFile := paths.ConfigFileWithDefault("otel.yml")
+			return validateOtelConfig(ctx, cfgFile)
+		},
+	}
+
+	return cmd
+}
+
+func validateOtelConfig(ctx context.Context, cfgFile string) error {
+	if runAsOtel := otel.IsOtelConfig(ctx, cfgFile); !runAsOtel {
+		return fmt.Errorf("%q is not an otel config. file should be named 'otel.yml', 'otlp.yml' or 'otelcol.yml'", cfgFile)
+	}
```

Comment:
After this PR is merged but before we change `elastic-agent validate` to `elastic-agent otel validate`, the user experience would be that `elastic-agent validate` only works for OTel configurations.  To me, this is a big awkward as `validate` is a top-level sub-command and, therefore, suggests that it can validate both OTel and Agent configurations.

So my suggestion for this PR would be to return early from this function with `nil` if `otel.IsOtelConfig(ctx, cfgFile)` returns `false` and also make `DefaultConfigName` the default value for the `-c/--config` global CLI options.

Once we have the follow up PR to move `elastic-agent validate` to `elastic-agent otel validate`, it would make sense to error out if `otel.IsOtelConfig(ctx, cfgFile)` returns false.